components/input_environmental_personal.py

```diff
@@ -366,52 +366,43 @@ def create_autocomplete(values: ModelInputsInfo):
     )
 
 
-@callback(
-    Output(ElementsIDs.met_input.value, "data"),
-    Output(ElementsIDs.met_input.value, "value"),
-    Input(ElementsIDs.met_input.value, "value"),
-    State(ElementsIDs.met_input.value, "data"),
-)
-def update_metabolic_rate_options(input_value, _):
+def update_options(input_value, options, selection_enum):
     if input_value is None or input_value == "":
         return [], ""
 
-    metabolic_rates = [option.value for option in MetabolicRateSelection]
+    option_values = [option.value for option in selection_enum]
 
-    # if the user has selected a predefined value from the dropdown
-    if input_value in metabolic_rates:
-        return metabolic_rates, float(input_value.split(":")[-1].strip().split()[0])
+    if input_value in option_values:
+        return option_values, float(input_value.split(":")[-1].strip().split()[0])
 
-    # if the user is typing text
     try:
         input_number = float(input_value)
+        filtered_options = [
+            option.value
+            for option in selection_enum
+            if abs(float(option.value.split(":")[-1].strip().split()[0]) - input_number)
+            < 1
```

Comment:
I never understood why we need <1

Reply:
This condition is used to find a "close" numeric match. It calculates the absolute difference between the input number and the value of each option. If the difference is less than 1, it is considered a match.
For example, if we enter "2".  The code: `walking_2mph: str = "Walking 2mph (3.2kmh): 2.0"` is included because |2.0 - 2| = 0 < 1

Also,we can modify the condition from '< 1' to a more stringent criterion, such as '<0.5'. This adjustment will provide more precise matching results. By lowering the threshold, we can narrow the matching range, returning only options that are closer to the input value.

This figure shows the threshold is 1.
![image](https://github.com/user-attachments/assets/1ac81d12-1ae8-4b3b-8d4d-c4317118fa12)

This figure shows the threshold is 0.5.
![image](https://github.com/user-attachments/assets/08cd6510-94d7-472c-a536-5901c6b9cdec)

Reply:
thank you for the explanation, what is not clear to me is then why we are only including values that are above 2 and not below 2. If the condition is 0.5 we should see all the activities in the range of 1.5 and 2.5.

On top of that can we also split the code in two lines, one in which we extract the value float(option.value.split(":")[-1].strip().split()[0] and one in which we do the comparison?

Reply:
Yes, if the condition is 0.5, the range is 1.5-2.5. But our input value is 2, which means the code will match the options like 1.2,2.1,2.2...,because the condition is 0.5, |1.2-2|=0.8>0.5 so this option will not be display. 

Summary:  The code first considers all values within the range of 1.5 to 2.5, which is calculated based on the input value of 2 and the threshold of 0.5. Within this range, the code further filters out all options that contain the digit "2". This means that values such as 2.0, 2.1, and 2.2 will be included in the results because they are not only within the specified range but also contain the digit "2". Moreover, if the option list includes values like 1.52 or 1.92, they would also be included in the result set for the same reasons: they fall within the range of 1.5 to 2.5 and contain the digit "2".

If you want to show all the activities in the range of 1.5 and 2.5 when the condition is 0.5, I can modift it. But i think the when the condition is 1, we can see all the activities because the range is [1,3].


For the last question : I can modify it to make it two lines.